### PR TITLE
Homework 6 (Alexander Ryzhickov)

### PR DESCRIPTION
**Pull request by AlexRyzhickov**
ok github.com/andreipimenov/golang-training-2021/internal/service 0.030s coverage: 84.7% of statements

### CODE REVIEW

File: internal/service/service_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+				}
+			}
+		}`,
+	dateString, price.Open, price.High, price.Low, price.Close)
+
+var price = model.Price{
+	Open:  "40.0",
+	High:  "40.0",
+	Low:   "40.0",
+	Close: "40.0",
```
Comment (subtle-byte) [inline internal/service/service_test.go:45]: Probably it is better to use different values to check they were not mixed up.

File: internal/service/service_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	suite.Equal(price, *p)
+}
+
+func (suite *serviceTestSuite) TestServiceWithoutCache() {
+
+	recorder := httptest.NewRecorder()
+	recorder.Code = http.StatusOK
+	recorder.Body.WriteString(validResponseJSON)
+
+	suite.repoMock.On("Load", validKeyRepo).Once().Return(price, false)
```
Comment (subtle-byte) [inline internal/service/service_test.go:85]: Actual `Load` will return `model.Price{}, false`. So it seems like `price` should be replaced by `model.Price{}` here.

File: internal/service/service_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	}
+
+	for index, value := range jsonFields {
+		resJson := strings.Replace(validResponseJSON, value, replace(index), 1)
+
+		recorder := httptest.NewRecorder()
+		recorder.Code = http.StatusOK
+		recorder.Body.WriteString(resJson)
+
+		suite.repoMock.On("Load", validKeyRepo).Once().Return(price, false)
```
Comment (subtle-byte) [inline internal/service/service_test.go:119]: The same.

File: internal/service/service_test.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+	jsonFields := [4]string{
+		fmt.Sprintf(`"1. open": "%s",`, price.Open),
+		fmt.Sprintf(`"2. high": "%s",`, price.High),
+		fmt.Sprintf(`"3. low": "%s",`, price.Low),
+		fmt.Sprintf(`"4. close": "%s"`, price.Close),
+	}
+
+	replace := func(index int) string {
+		if index == len(jsonFields) {
```
Comment (subtle-byte) [inline internal/service/service_test.go:106]: Seems like it never can be true.

File: internal/service/service_test.go [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+		_, err := suite.service.GetPrice(validTicker, date)
+		suite.Error(err)
+	}
+}
+
+func (suite *serviceTestSuite) TestHTTPClient() {
+	suite.repoMock.
+		On("Load", validKeyRepo).
+		Once().
+		Return(price, false)
```
Comment (subtle-byte) [inline internal/service/service_test.go:133]: The same.

**Review by subtle-byte — COMMENTED** [review]
Anyway LGTM.

**Comment by pavlov-tony** [comment]
@AlexRyzhickov, I'll close this PR, but feel free to address all comments was left.
@subtle-byte great review!

**Pull request closed without merging**